cogs/ahk/ahk.py

```diff
@@ -944,6 +947,165 @@ async def on_member_remove(self, member: disnake.Member):
+            except disnake.HTTPException:
+                return
+        else:
+            message: disnake.Message = thing
+            if not (
+                message.channel.id in HELP_CHANNEL_IDS
+                or isinstance(message.channel, disnake.Thread)
+            ):
+                return
+
+        if not message.attachments:
+            return
+
+        async def _parse_attachments(attachments: list[disnake.Attachment]) -> list[str, str]:
+            valid_attachments = []
+            for attachment in attachments:
+                if attachment.size > 2 * 1024 * 1024:
+                    continue
+
+                try:
+                    data = await attachment.read()
+                except (disnake.HTTPException, disnake.NotFound):
+                    continue
+
+                try:
+                    decoded = data.decode("utf-8-sig")
+                except UnicodeDecodeError:
+                    continue
+
+                if "\0" in decoded:
+                    continue
+
+                valid_attachments.append([attachment.filename, decoded])
+            return valid_attachments
+
+        valid_attachments = await _parse_attachments(message.attachments)
+
+        if not valid_attachments:
+            return
+
+        embed = disnake.Embed()
+        embed.title = "Share your file(s) using our paste service (p.autohotkey.com)?"
+        embed.description = (
+            "Click Yes if you want to upload them automatically.\n"
+            "This allows others to read your code without downloading files."
+        )
+        embed.color = disnake.Color.blue()
+
+        row = disnake.ui.ActionRow()
+
+        row.add_button(
+            style=disnake.ButtonStyle.primary,
+            custom_id="pahk_upload",
+            label="Upload",
+            emoji="☁️",
+        )
+
+        reply = await message.reply(embed=embed, components=[row])
+
+        def iter_checker(
+            author: disnake.Member | disnake.User,
+            channel: disnake.GroupChannel,
+            expected_custom_id: str,
+        ):
+            def check(inter: disnake.MessageInteraction) -> bool:
+                if inter.author != author:
+                    return False
+                if inter.channel != channel:
+                    return False
+                return (
+                    hasattr(inter.component, "custom_id")
+                    and inter.component.custom_id == expected_custom_id
+                )
+
+            return check
+
+        async def upload_to_pahk(attachments: list[str, str]) -> list[str, str]:
+            url = "https://pahk.geekdude.io/"
+            headers = {"Content-Type": "application/x-www-form-urlencoded"}
+            links = []
+            async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(ssl=False)) as session:
+                for filename, attachment in attachments:
+                    try:
+                        async with session.post(
```

Comment:
you can run all uploads in in parallel in an `asyncio.gather` if you wanna get really fancy. but not needed